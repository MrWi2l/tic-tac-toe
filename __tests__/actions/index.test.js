import {
  initializeSquareAction,
  initializePlayerAction,
  addSquareAction,
  changePlayerAction,
} from '../../src/actions/index'

describe('ACTIONS', () => {
  it('Should create an action to initialize squares', () => {
    expect(initializeSquareAction()).toMatchSnapshot()
  })

  it('Should create an action to initialize player', () => {
    expect(initializePlayerAction()).toMatchSnapshot()
  })

  it('Should create an action to add square', () => {
    const wrapper = (index, symbol) => ({
      type: 'ADD_SQUARE',
      index,
      symbol,
    })
    expect(addSquareAction(1, 'X')).toEqual(wrapper(1, 'X'))
    expect(addSquareAction(1, 'X')).toMatchSnapshot()
  })

  it('Should create an action to change player', () => {
    const wrapper = currentPlayer => ({
      type: 'CHANGE_PLAYER',
      currentPlayer,
    })
    expect(changePlayerAction('X')).toEqual(wrapper('X'))
    expect(changePlayerAction('X')).toMatchSnapshot()
  })
})
