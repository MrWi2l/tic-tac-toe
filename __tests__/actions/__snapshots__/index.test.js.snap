exports[`ACTIONS Should create an action to add square 1`] = `
Object {
  "index": 1,
  "symbol": "X",
  "type": "ADD_SQUARE",
}
`;

exports[`ACTIONS Should create an action to change player 1`] = `
Object {
  "currentPlayer": "X",
  "type": "CHANGE_PLAYER",
}
`;

exports[`ACTIONS Should create an action to initialize player 1`] = `
Object {
  "type": "INITIALIZE_PLAYER",
}
`;

exports[`ACTIONS Should create an action to initialize squares 1`] = `
Object {
  "type": "INITIALIZE_SQUARES",
}
`;
